File: Python/ByTier/A/0586_A_Alena_s_Schedule.py
# Contest 586, Problem A: Alena's Schedule
# URL: https://codeforces.com/contest/586/problem/A

n = int(input())
word = input().replace(" ", "").strip("0")
word = word.replace("0"*5, "00")
while "000" in word:
    word = word.replace("000", "00")
word = word.replace("00", "")
print(len(word))

File: CPP/ByRound/0586/0586_A_Alena_s_Schedule.cpp
/*
 * Problem URL : https://codeforces.com/contest/586/problem/A
 * Submit Date : 2025-09-04
 */

#include <bits/stdc++.h>
using namespace std;
int k[101];
int main() {
  int n, i, z = 0;
  cin >> n;
  for (i = 1; i <= n; i++) {
    cin >> k[i];
    if (k[i] == 1)
      z++;
  }
  for (i = 2; i < n; i++)
    if (k[i] == 0 && k[i - 1] == 1 && k[i + 1] == 1)
      z++;
  cout << z;
}